--- a/extracted-drivers/panel-cwu50/panel-cwu50.c
+++ b/extracted-drivers/panel-cwu50/panel-cwu50.c
@@ -7,7 +7,8 @@
 #include <linux/gpio/consumer.h>
 #include <linux/regulator/consumer.h>
 #include <linux/delay.h>
-#include <linux/of_device.h>
+#include <linux/of.h>
+#include <linux/mod_devicetable.h>
 #include <linux/module.h>
 
 struct cwu50 {
@@ -685,9 +686,6 @@
 	connector->display_info.width_mm = mode->width_mm;
 	connector->display_info.height_mm = mode->height_mm;
 
-	/* set up connector's "panel orientation" property */
-	drm_connector_set_panel_orientation(connector, ctx->orientation);
-
 	drm_mode_probed_add(connector, mode);
 
 	return 1; /* Number of modes */
