--- a/extracted-drivers/panel-cwu50/panel-cwu50.c
+++ b/extracted-drivers/panel-cwu50/panel-cwu50.c
@@ -7,7 +7,8 @@
 #include <linux/gpio/consumer.h>
 #include <linux/regulator/consumer.h>
 #include <linux/delay.h>
-#include <linux/of_device.h>
+#include <linux/of.h>
+#include <linux/mod_devicetable.h>
 #include <linux/module.h>
 
 struct cwu50 {
@@ -24,15 +25,15 @@
 };
 
 static const struct drm_display_mode default_mode = {
-	.clock = 61020,
+	.clock = 40000,
 	.hdisplay = 720,
-	.hsync_start = 720 + 30,
-	.hsync_end = 720+ 30 + 15,
-	.htotal = 720 + 30 + 15 + 15,
+	.hsync_start = 720 + 80,
+	.hsync_end = 720 + 80 + 40,
+	.htotal = 720 + 80 + 40 + 80,
 	.vdisplay = 1280,
-	.vsync_start = 1280 + 8,
-	.vsync_end = 1280 + 8+ 2,
-	.vtotal = 1280 + 8 + 2 + 16,
+	.vsync_start = 1280 + 40,
+	.vsync_end = 1280 + 40 + 10,
+	.vtotal = 1280 + 40 + 10 + 40,
 };
 
 static inline struct cwu50 *panel_to_cwu50(struct drm_panel *panel)
@@ -268,14 +269,14 @@
 	int err;
 	dcs_write_seq(0xE0,0x00);
 
-	//--- PASSWORD	----//
+	//--- PASSWORD	----
 	dcs_write_seq(0xE1,0x93);
 	dcs_write_seq(0xE2,0x65);
 	dcs_write_seq(0xE3,0xF8);
 	dcs_write_seq(0x80,0x03);//03:4lane 02:3lane 01:2lane
 
 
-	//--- Page1  ----//
+	//--- Page1  ----
 	dcs_write_seq(0xE0,0x01);
 
 	//Set VCOM
@@ -296,20 +297,20 @@
 	dcs_write_seq(0x24,0xFE);
 
 	//Set Panel
-	dcs_write_seq(0x37,0x09);	//SS=1,BGR=1
+	dcs_write_seq(0x37,0x09); //SS=1,BGR=1
 
 	//SET RGBCYC
-	dcs_write_seq(0x38,0x04);	//JDT=100 column inversion
-	dcs_write_seq(0x39,0x08);	//RGB_N_EQ1
-	dcs_write_seq(0x3A,0x12);	//RGB_N_EQ2
-	dcs_write_seq(0x3C,0x78);	//SET EQ3 for TE_H
+	dcs_write_seq(0x38,0x04); //JDT=100 column inversion
+	dcs_write_seq(0x39,0x08); //RGB_N_EQ1
+	dcs_write_seq(0x3A,0x12); //RGB_N_EQ2
+	dcs_write_seq(0x3C,0x78); //SET EQ3 for TE_H
 	dcs_write_seq(0x3D,0xFF);
 	dcs_write_seq(0x3E,0xFF);
 	dcs_write_seq(0x3F,0xFF);
 
 	//Set TCON
-	dcs_write_seq(0x40,0x04);	//RSO 04h=720, 05h=768, 06h=800
-	dcs_write_seq(0x41,0xA0);	//LN=640->1280 line
+	dcs_write_seq(0x40,0x04); //RSO 04h=720, 05h=768, 06h=800
+	dcs_write_seq(0x41,0xA0); //LN=640->1280 line
 	dcs_write_seq(0x42,0x7F);  //SLT=12.7us
 	dcs_write_seq(0x43,0x10);  //VFP
 	dcs_write_seq(0x44,0x17);  //VBP =24
@@ -317,16 +318,16 @@
 
 	//dcs_write_seq(0x4A,0x35);//BIST MODE 35:AUTO
 
-	//--- power voltage  ----//
-	dcs_write_seq(0x55,0x02);	//DCDCM=0011, JD5001
+	//--- power voltage  ----
+	dcs_write_seq(0x55,0x02); //DCDCM=0011, JD5001
 	//dcs_write_seq(0x56,0x01);
 	dcs_write_seq(0x57,0x69);
 	//dcs_write_seq(0x58,0x0A);
-	dcs_write_seq(0x59,0x2A);	//VCL = -2.7V, AVEE=-5.5V
-	dcs_write_seq(0x5A,0x1A);	//VGH = +12.2V
-	dcs_write_seq(0x5B,0x1A);	//VGL = -12.2V
+	dcs_write_seq(0x59,0x2A); //VCL = -2.7V, AVEE=-5.5V
+	dcs_write_seq(0x5A,0x1A); //VGH = +12.2V
+	dcs_write_seq(0x5B,0x1A); //VGL = -12.2V
 
-	//--- Gamma2.2	----//	//G2.2	  //G2.5
+	//--- Gamma2.2	----	//G2.2	  //G2.5
 	dcs_write_seq(0x5D,0x7F);  //0x7F	//0x7F
 	dcs_write_seq(0x5E,0x67);  //0x67	//0x65
 	dcs_write_seq(0x5F,0x58);  //0x58	//0x55
@@ -371,100 +372,100 @@
 	dcs_write_seq(0xE0,0x02);
 
 	//GIP_L Pin mapping
-	dcs_write_seq(0x00,0x5F);	//GCL
-	dcs_write_seq(0x01,0x5F);	//VSS->VGL
-	dcs_write_seq(0x02,0x44);	//CLK1->CKV0
-	dcs_write_seq(0x03,0x46);	//CKK3->CKV2
-	dcs_write_seq(0x04,0x48);	//CLK5->CKV4
-	dcs_write_seq(0x05,0x4A);	//CLK7->CKV6
-	dcs_write_seq(0x06,0x5F);	//VGL
-	dcs_write_seq(0x07,0x5F);	//VGL
-	dcs_write_seq(0x08,0x5F);	//VGL
-	dcs_write_seq(0x09,0x5F);	//NC
-	dcs_write_seq(0x0A,0x5F);	//NC
-	dcs_write_seq(0x0B,0x5F);	//NC //
-	dcs_write_seq(0x0C,0x5F);	//NC
-	dcs_write_seq(0x0D,0x5F);	//NC
-	dcs_write_seq(0x0E,0x5F);	//NC //
-	dcs_write_seq(0x0F,0x5F);	//NC
-	dcs_write_seq(0x10,0x5F);	//NC
-	dcs_write_seq(0x11,0x5F);	//NC //
-	dcs_write_seq(0x12,0x5E);	//GCH
-	dcs_write_seq(0x13,0x5E);	//VDD->VGH
-	dcs_write_seq(0x14,0x40);	//STV1
-	dcs_write_seq(0x15,0x42);	//STV3
+	dcs_write_seq(0x00,0x5F); //GCL
+	dcs_write_seq(0x01,0x5F); //VSS->VGL
+	dcs_write_seq(0x02,0x44); //CLK1->CKV0
+	dcs_write_seq(0x03,0x46); //CKK3->CKV2
+	dcs_write_seq(0x04,0x48); //CLK5->CKV4
+	dcs_write_seq(0x05,0x4A); //CLK7->CKV6
+	dcs_write_seq(0x06,0x5F); //VGL
+	dcs_write_seq(0x07,0x5F); //VGL
+	dcs_write_seq(0x08,0x5F); //VGL
+	dcs_write_seq(0x09,0x5F); //NC
+	dcs_write_seq(0x0A,0x5F); //NC
+	dcs_write_seq(0x0B,0x5F); //NC //
+	dcs_write_seq(0x0C,0x5F); //NC
+	dcs_write_seq(0x0D,0x5F); //NC
+	dcs_write_seq(0x0E,0x5F); //NC //
+	dcs_write_seq(0x0F,0x5F); //NC
+	dcs_write_seq(0x10,0x5F); //NC
+	dcs_write_seq(0x11,0x5F); //NC //
+	dcs_write_seq(0x12,0x5E); //GCH
+	dcs_write_seq(0x13,0x5E); //VDD->VGH
+	dcs_write_seq(0x14,0x40); //STV1
+	dcs_write_seq(0x15,0x42); //STV3
 
 	//GIP_R Pin mapping
-	dcs_write_seq(0x16,0x5F);	//GCL
-	dcs_write_seq(0x17,0x5F);	//VSS->VGL
-	dcs_write_seq(0x18,0x45);	//CLK2->CKV1
-	dcs_write_seq(0x19,0x47);	//CKK4->CKV3
-	dcs_write_seq(0x1A,0x49);	//CLK6->CKV5
-	dcs_write_seq(0x1B,0x4B);	//CLK8->CKV7
-	dcs_write_seq(0x1C,0x5F);	//VGL
-	dcs_write_seq(0x1D,0x5F);	//VGL
-	dcs_write_seq(0x1E,0x5F);	//VGL
-	dcs_write_seq(0x1F,0x5F);	//NC
-	dcs_write_seq(0x20,0x5F);	//NC
-	dcs_write_seq(0x21,0x5F);	//NC //
-	dcs_write_seq(0x22,0x5F);	//NC
-	dcs_write_seq(0x23,0x5F);	//NC
-	dcs_write_seq(0x24,0x5F);	//NC //
-	dcs_write_seq(0x25,0x5F);	//NC
-	dcs_write_seq(0x26,0x5F);	//NC
-	dcs_write_seq(0x27,0x5F);	//NC //
-	dcs_write_seq(0x28,0x5E);	//GCH
-	dcs_write_seq(0x29,0x5E);	//VDD->VGH
-	dcs_write_seq(0x2A,0x41);	//STV2
-	dcs_write_seq(0x2B,0x43);	//STV4
+	dcs_write_seq(0x16,0x5F); //GCL
+	dcs_write_seq(0x17,0x5F); //VSS->VGL
+	dcs_write_seq(0x18,0x45); //CLK2->CKV1
+	dcs_write_seq(0x19,0x47); //CKK4->CKV3
+	dcs_write_seq(0x1A,0x49); //CLK6->CKV5
+	dcs_write_seq(0x1B,0x4B); //CLK8->CKV7
+	dcs_write_seq(0x1C,0x5F); //VGL
+	dcs_write_seq(0x1D,0x5F); //VGL
+	dcs_write_seq(0x1E,0x5F); //VGL
+	dcs_write_seq(0x1F,0x5F); //NC
+	dcs_write_seq(0x20,0x5F); //NC
+	dcs_write_seq(0x21,0x5F); //NC //
+	dcs_write_seq(0x22,0x5F); //NC
+	dcs_write_seq(0x23,0x5F); //NC
+	dcs_write_seq(0x24,0x5F); //NC //
+	dcs_write_seq(0x25,0x5F); //NC
+	dcs_write_seq(0x26,0x5F); //NC
+	dcs_write_seq(0x27,0x5F); //NC //
+	dcs_write_seq(0x28,0x5E); //GCH
+	dcs_write_seq(0x29,0x5E); //VDD->VGH
+	dcs_write_seq(0x2A,0x41); //STV2
+	dcs_write_seq(0x2B,0x43); //STV4
 
 	//GIP_L_GS Pin mapping
-	dcs_write_seq(0x2C,0x1F);	//GCL
-	dcs_write_seq(0x2D,0x1E);	//VSS->VGH
-	dcs_write_seq(0x2E,0x0B);	//CLK1->CKV7
-	dcs_write_seq(0x2F,0x09);	//CKK3->CKV5
-	dcs_write_seq(0x30,0x07);	//CLK5->CKV3
-	dcs_write_seq(0x31,0x05);	//CLK7->CKV1
-	dcs_write_seq(0x32,0x1F);	//VGL
-	dcs_write_seq(0x33,0x1F);	//VGL
-	dcs_write_seq(0x34,0x1F);	//VGL
-	dcs_write_seq(0x35,0x1F);	//NC
-	dcs_write_seq(0x36,0x1F);	//NC
-	dcs_write_seq(0x37,0x1F);	//NC //
-	dcs_write_seq(0x38,0x1F);	//NC
-	dcs_write_seq(0x39,0x1F);	//NC
-	dcs_write_seq(0x3A,0x1F);	//NC //
-	dcs_write_seq(0x3B,0x1F);	//NC
-	dcs_write_seq(0x3C,0x1F);	//NC
-	dcs_write_seq(0x3D,0x1F);	//NC //
-	dcs_write_seq(0x3E,0x1E);	//GCH
-	dcs_write_seq(0x3F,0x1F);	//VDD->VGL
-	dcs_write_seq(0x40,0x03);	//STV1
-	dcs_write_seq(0x41,0x01);	//STV3
+	dcs_write_seq(0x2C,0x1F); //GCL
+	dcs_write_seq(0x2D,0x1E); //VSS->VGH
+	dcs_write_seq(0x2E,0x0B); //CLK1->CKV7
+	dcs_write_seq(0x2F,0x09); //CKK3->CKV5
+	dcs_write_seq(0x30,0x07); //CLK5->CKV3
+	dcs_write_seq(0x31,0x05); //CLK7->CKV1
+	dcs_write_seq(0x32,0x1F); //VGL
+	dcs_write_seq(0x33,0x1F); //VGL
+	dcs_write_seq(0x34,0x1F); //VGL
+	dcs_write_seq(0x35,0x1F); //NC
+	dcs_write_seq(0x36,0x1F); //NC
+	dcs_write_seq(0x37,0x1F); //NC //
+	dcs_write_seq(0x38,0x1F); //NC
+	dcs_write_seq(0x39,0x1F); //NC
+	dcs_write_seq(0x3A,0x1F); //NC //
+	dcs_write_seq(0x3B,0x1F); //NC
+	dcs_write_seq(0x3C,0x1F); //NC
+	dcs_write_seq(0x3D,0x1F); //NC //
+	dcs_write_seq(0x3E,0x1E); //GCH
+	dcs_write_seq(0x3F,0x1F); //VDD->VGL
+	dcs_write_seq(0x40,0x03); //STV1
+	dcs_write_seq(0x41,0x01); //STV3
 
 	//GIP_R_GS Pin mapping
-	dcs_write_seq(0x42,0x1F);	//GCL
-	dcs_write_seq(0x43,0x1E);	//VSS->VGH
-	dcs_write_seq(0x44,0x0A);	//CLK2->CKV6
-	dcs_write_seq(0x45,0x08);	//CKK4->CKV4
-	dcs_write_seq(0x46,0x06);	//CLK6->CKV2
-	dcs_write_seq(0x47,0x04);	//CLK8->CKV0
-	dcs_write_seq(0x48,0x1F);	//VGL
-	dcs_write_seq(0x49,0x1F);	//VGL
-	dcs_write_seq(0x4A,0x1F);	//VGL
-	dcs_write_seq(0x4B,0x1F);	//NC
-	dcs_write_seq(0x4C,0x1F);	//NC
-	dcs_write_seq(0x4D,0x1F);	//NC //
-	dcs_write_seq(0x4E,0x1F);	//NC
-	dcs_write_seq(0x4F,0x1F);	//NC
-	dcs_write_seq(0x50,0x1F);	//NC //
-	dcs_write_seq(0x51,0x1F);	//NC
-	dcs_write_seq(0x52,0x1F);	//NC
-	dcs_write_seq(0x53,0x1F);	//NC //
-	dcs_write_seq(0x54,0x1E);	//GCH
-	dcs_write_seq(0x55,0x1F);	//VDD->VGL
-	dcs_write_seq(0x56,0x02);	//STV2
-	dcs_write_seq(0x57,0x00);	//STV4
+	dcs_write_seq(0x42,0x1F); //GCL
+	dcs_write_seq(0x43,0x1E); //VSS->VGH
+	dcs_write_seq(0x44,0x0A); //CLK2->CKV6
+	dcs_write_seq(0x45,0x08); //CKK4->CKV4
+	dcs_write_seq(0x46,0x06); //CLK6->CKV2
+	dcs_write_seq(0x47,0x04); //CLK8->CKV0
+	dcs_write_seq(0x48,0x1F); //VGL
+	dcs_write_seq(0x49,0x1F); //VGL
+	dcs_write_seq(0x4A,0x1F); //VGL
+	dcs_write_seq(0x4B,0x1F); //NC
+	dcs_write_seq(0x4C,0x1F); //NC
+	dcs_write_seq(0x4D,0x1F); //NC //
+	dcs_write_seq(0x4E,0x1F); //NC
+	dcs_write_seq(0x4F,0x1F); //NC
+	dcs_write_seq(0x50,0x1F); //NC //
+	dcs_write_seq(0x51,0x1F); //NC
+	dcs_write_seq(0x52,0x1F); //NC
+	dcs_write_seq(0x53,0x1F); //NC //
+	dcs_write_seq(0x54,0x1E); //GCH
+	dcs_write_seq(0x55,0x1F); //VDD->VGL
+	dcs_write_seq(0x56,0x02); //STV2
+	dcs_write_seq(0x57,0x00); //STV4
 
 	//GIP Timing
 	dcs_write_seq(0x58,0x40);
@@ -524,7 +525,7 @@
 	msleep (100);
 
 
-	//--- TE----//
+	//--- TE----
 	dcs_write_seq(0x35,0x00);
 
 	return 0;
@@ -533,8 +534,6 @@
 static int cwu50_disable(struct drm_panel *panel)
 {
 	struct cwu50 *ctx = panel_to_cwu50(panel);
-	struct mipi_dsi_device *dsi = to_mipi_dsi_device(ctx->dev);
-	int ret;
 
 	if (!ctx->enabled)
 		return 0;
@@ -585,7 +584,6 @@
 	struct cwu50 *ctx = panel_to_cwu50(panel);
 	struct mipi_dsi_device *dsi = to_mipi_dsi_device(ctx->dev);
 	int ret;
-	u8 buf[4];
 
 	if (ctx->prepared)
 		return 0;
@@ -621,9 +619,9 @@
 		dev_info(ctx->dev, "old panel, cycling the reset pin\n");
 		/* Cycle RESX (Hardware Reset) */
 		gpiod_set_value_cansleep(ctx->id_gpio, 1);
-		msleep(10);
+		msleep(50);
 		gpiod_set_value_cansleep(ctx->id_gpio, 0);
-		msleep(5);
+		msleep(20);
 	}
 
 	/* Enabe tearing mode: send TE (tearing effect) at VBLANK */
@@ -633,13 +631,6 @@
 		return ret;
 	}
 	/* Exit sleep mode and power on */
-	dcs_write_seq(0x11);// SLPOUT
-        msleep(120);
-        dcs_write_seq(0xE0,0x00);
-        mipi_dsi_dcs_read(dsi, 0x04, buf, 3);
-	
-        if(buf[0] == 0x39) ctx->is_new_panel = 1;
-
 	if (ctx->is_new_panel)
 		cwu50_init_sequence2(ctx);
 	else
@@ -657,7 +648,7 @@
 		dev_err(ctx->dev, "failed to turn display on (%d)\n", ret);
 		return ret;
 	}
-	msleep(20);
+	msleep(100);
 
 	ctx->prepared = true;
 
@@ -667,8 +658,6 @@
 static int cwu50_enable(struct drm_panel *panel)
 {
 	struct cwu50 *ctx = panel_to_cwu50(panel);
-	struct mipi_dsi_device *dsi = to_mipi_dsi_device(ctx->dev);
-	int ret;
 
 	if (ctx->enabled)
 		return 0;
@@ -696,9 +685,6 @@
 	connector->display_info.width_mm = mode->width_mm;
 	connector->display_info.height_mm = mode->height_mm;
 
-	/* set up connector's "panel orientation" property */
-	drm_connector_set_panel_orientation(connector, ctx->orientation);
-
 	drm_mode_probed_add(connector, mode);
 
 	return 1; /* Number of modes */
@@ -718,16 +704,19 @@
 	struct cwu50 *ctx;
 	int ret, err;
 
+	dev_info(dev, "PROBE START: Hardcoding Right-Up orientation\n");
+
 	ctx = devm_kzalloc(dev, sizeof(*ctx), GFP_KERNEL);
 	if (!ctx)
 		return -ENOMEM;
 
+	ctx->orientation = DRM_MODE_PANEL_ORIENTATION_RIGHT_UP;
 	mipi_dsi_set_drvdata(dsi, ctx);
 	ctx->dev = dev;
 
 	dsi->lanes = 4;
 	dsi->format = MIPI_DSI_FMT_RGB888;
-	dsi->mode_flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST | MIPI_DSI_MODE_VIDEO_SYNC_PULSE;
+	dsi->mode_flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST;
 
 	ctx->id_gpio = devm_gpiod_get_optional(dev, "reset", GPIOD_IN);
 	if (IS_ERR(ctx->id_gpio)) {
@@ -737,12 +726,16 @@
 		return ret;
 	}
 
-	ctx->is_new_panel = gpiod_get_value_cansleep(ctx->id_gpio);
+	ctx->is_new_panel = false;
+	dev_info(dev, "Forced old panel type for reliable CM5 reset\n");
+
+	/*
 	if (ctx->is_new_panel) {
 		dev_info(dev, "Detected new panel type\n");
 	} else {
 		dev_info(dev, "Detected old panel type\n");
 	}
+	*/
 
 	/*
 	 * Switch the ID GPIO to OUTPUT for use with resetting,
@@ -796,14 +789,8 @@
 		return PTR_ERR(ctx->backlight);
 	}
 
-	ret = of_drm_get_panel_orientation(dev->of_node, &ctx->orientation);
-	if (ret) {
-		dev_err(dev, "%pOF: failed to get orientation %d\n", dev->of_node, ret);
-		return ret;
-	}
-
-	ctx->panel.prepare_prev_first = true;
-
+	ctx->orientation = DRM_MODE_PANEL_ORIENTATION_LEFT_UP;
+	dev_info(dev, "Forced Left-Up orientation for CM5\n");
 	drm_panel_init(&ctx->panel, dev, &cwu50_drm_funcs, DRM_MODE_CONNECTOR_DSI);
 
 	drm_panel_add(&ctx->panel);
